app/uk/gov/hmrc/exports/services/mapping/CachingMappingHelper.scala

```diff
@@ -51,11 +51,16 @@ class CachingMappingHelper {
         taricCodes.map(code => Classification(Some(code), identificationTypeCode = Some(TARICAdditionalCode.value)))
     ).toSeq
 
-  def mapGoodsMeasure(data: CommodityMeasure) =
-    Commodity(
-      goodsMeasure =
-        Some(GoodsMeasure(Some(createMeasure(data.grossMass)), Some(createMeasure(data.netMass)), data.supplementaryUnits.map(createMeasure)))
-    )
+  def mapGoodsMeasure(data: CommodityMeasure): Option[Commodity] = data match {
+    case CommodityMeasure(None, None, None) => None
+    case _ =>
```

Comment:
My propose is to change it to below. To extract values during pattern matching. 

```
case CommodityMeasure(supplementaryUnits, netMass, grossMass) =>
  Some(
    Commodity(
          goodsMeasure =
            Some(GoodsMeasure(grossMass.map(createMeasure), netMass.map(createMeasure), supplementaryUnits.map(createMeasure)))
        )
  )
```

Reply:
I thought of doing that but couldn't see any real advantage.  But in the spirit of the times I will update.

Reply:
Using pattern matching just to check if all the values are empty and after that come back to using the element that you're matching is not a good idea. For this approach is better to check something with `if` or prepare a method inside the `CommodityMeasure` class to check if it's empty.

Reply:
I've done it.   But I originally did it wrong.  I matched 
`case CommodityMeasure(grossMass, netMass, supplementaryUnits)`
i.e. I defined the variables in the order I was later using them.   It compiled (of course as they are all the same type).  Only spotted the error when a test failed.

So on reflection, my original way was less error prone.  IMO.

Reply:
Agree with you @SteveSugden but still I think it's not the reason of the using pattern matching :)